internal/services/identity_permissions.go

```diff
@@ -0,0 +1,111 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+)
+
+// HasVehiclePermissions checks if the given developer license has privileges 1,3,4 for the vehicle.
+func HasVehiclePermissions(identityAPIURL, vehicleTokenID string, devLicense []byte, logger zerolog.Logger) (bool, error) {
```

Comment:
I would consider having a wrapper class around identity-api, something [like this](https://github.com/DIMO-Network/valuations-api/blob/9aa0785bcd717cb2c8a2654ca9cf08be580d62e6/internal/core/gateways/identity_api.go) is a recent example.
Then, that can be mocked for testing, and bring in other places where you call out to identity under one place (eg. VerifyDeveloperLicense), or expanded for future cases where you need to call out to identity-api. I think it will help make the implementation a little cleaner too and simplify testing.